wodles/azure/azure-logs.py

```diff
@@ -602,33 +608,58 @@ def start_storage():
-
-        # Get the blobs
-        for container in containers:
-            md5_hash = md5(name.encode()).hexdigest()
-            offset = args.storage_time_offset
-            min_str, max_str = get_min_max(service_name="storage", md5_hash=md5_hash, offset=offset)
-            # "parse" adds compatibility with "last_dates_files" from previous releases as the format wasn't localized
-            # It also handles any datetime with more than 6 digits for the microseconds value provided by Azure
-            min_datetime = parse(min_str, fuzzy=True)
-            max_datetime = parse(max_str, fuzzy=True)
-            desired_datetime = offset_to_datetime(offset) if offset else max_datetime
-            name = container.name if args.container == '*' else args.container
+    block_blob_service = BlockBlobService(account_name=name, account_key=key)
+
+    # Disable max retry value before attempting to validate the credentials
+    old_retry_value = block_blob_service.retry
+    block_blob_service.retry = no_retry
+
+    # Verify if the credentials grant access to the specified container
+    if args.container != '*':
+        try:
+            if not block_blob_service.exists(args.container):
+                logging.error(f"Storage: The '{args.container}' container does not exists.")
+                sys.exit(1)
+            containers = [args.container]
+        except AzureException:
+            logging.error(f"Storage: Invalid credentials for accessing the '{args.container}' container.")
+            sys.exit(1)
+    else:
+        try:
+            logging.info("Storage: Getting containers.")
+            containers = block_blob_service.list_containers()
+        except AzureSigningError:
+            logging.error("Storage: Unable to list the containers. Invalid credentials.")
+            sys.exit(1)
+        except AzureException as e:
+            logging.error(f"Storage: The containers could not be listed: '{e.error_code}'.")
+            sys.exit(1)
+
+    # Restore the default max retry value
+    block_blob_service.retry = old_retry_value
+    logging.info("Storage: Authenticated.")
+
+    # Get the blobs
+    for container in containers:
+        md5_hash = md5(name.encode()).hexdigest()
+        # check if the hash was already present in the last_dates file
+        md5_exists = md5_hash in dates_json["storage"]
+        offset = args.storage_time_offset
+        min_str, max_str = get_min_max(service_name="storage", md5_hash=md5_hash, offset=offset)
+        # "parse" adds compatibility with "last_dates_files" from previous releases as the format wasn't localized
+        # It also handles any datetime with more than 6 digits for the microseconds value provided by Azure
+        min_datetime = parse(min_str, fuzzy=True)
+        max_datetime = parse(max_str, fuzzy=True)
+        desired_datetime = offset_to_datetime(offset) if offset else max_datetime
+        name = container.name if args.container == '*' else args.container
```

Comment:
At this point containers can be a either a list of string with one element or a list of `ContainerProperties`. Both should be a list of `str` with the names of the containers, this would be less error-prone and would make the last assignment unnecessary.

Reply:
Agreed. The code has been updated to ensure the list will be always composed of str values